src/libraries/Microsoft.PowerFx.Connectors/ArgumentMapper.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AppMagic.Authoring;
+using Microsoft.OpenApi.Models;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+using Contracts = Microsoft.PowerFx.Core.Utils.Contracts;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    // Map arguments between a Swagger File and a Function signature. 
+    // Deals with: 
+    //  - internals, "connectionId"
+    //  - optional, required, default value 
+    //  - computing signature type
+    //
+    // Given a swagger definition, handles:
+    //  - signature information to create a ServiceFunction
+    //  - reverse mapping to map from function signature back to swagger.
+    internal class ArgumentMapper
```

Comment:
>ArgumentMapper [](http://example.com/codeflow?start=19&length=14)

This class is critical. 
I really wish existing PAClient code had something like this where the logic was all centralized in one place.